trace/tests/system/gapic/v1/test_system_trace_service_v1_vpcsc.py

```diff
@@ -21,68 +21,52 @@
-PROJECT_OUTSIDE = os.environ.get(
-    "GOOGLE_CLOUD_TESTS_VPCSC_OUTSIDE_PERIMETER_PROJECT", None
-)
-
-
-class TestVPCServiceControlV1(object):
-    @staticmethod
-    def _is_rejected(call):
-        try:
-            responses = call()
-        except exceptions.PermissionDenied as e:
-            return e.message == "Request is prohibited by organization's policy"
-        except:
-            return False
-        return False
-
-    @pytest.mark.skipif(
-        PROJECT_INSIDE is None, reason="Missing environment variable: PROJECT_ID"
-    )
-    @pytest.mark.skipif(
-        PROJECT_OUTSIDE is None,
-        reason="Missing environment variable: GOOGLE_CLOUD_TESTS_VPCSC_OUTSIDE_PERIMETER_PROJECT",
-    )
-    def test_list_traces(self):
-        client = trace_v1.TraceServiceClient()
-
-        list_inside = lambda: list(client.list_traces(PROJECT_INSIDE))
-        list_outside = lambda: list(client.list_traces(PROJECT_OUTSIDE))
-
-        assert not TestVPCServiceControlV1._is_rejected(list_inside)
-        assert TestVPCServiceControlV1._is_rejected(list_outside)
-
-    @pytest.mark.skipif(
-        PROJECT_INSIDE is None, reason="Missing environment variable: PROJECT_ID"
-    )
-    @pytest.mark.skipif(
-        PROJECT_OUTSIDE is None,
-        reason="Missing environment variable: GOOGLE_CLOUD_TESTS_VPCSC_OUTSIDE_PERIMETER_PROJECT",
-    )
-    def test_get_trace(self):
-        client = trace_v1.TraceServiceClient()
-
-        get_inside = lambda: client.get_trace(PROJECT_INSIDE, "")
-        get_outside = lambda: client.get_trace(PROJECT_OUTSIDE, "")
-
-        assert not TestVPCServiceControlV1._is_rejected(get_inside)
-        assert TestVPCServiceControlV1._is_rejected(get_outside)
-
-    @pytest.mark.skipif(
-        PROJECT_INSIDE is None, reason="Missing environment variable: PROJECT_ID"
-    )
-    @pytest.mark.skipif(
-        PROJECT_OUTSIDE is None,
-        reason="Missing environment variable: GOOGLE_CLOUD_TESTS_VPCSC_OUTSIDE_PERIMETER_PROJECT",
-    )
-    def test_patch_traces(self):
-        client = trace_v1.TraceServiceClient()
-
-        patch_inside = lambda: client.patch_traces(PROJECT_INSIDE, {})
-        patch_outside = lambda: client.patch_traces(PROJECT_OUTSIDE, {})
-
-        assert not TestVPCServiceControlV1._is_rejected(patch_inside)
-        assert TestVPCServiceControlV1._is_rejected(patch_outside)
+_VPCSC_PROHIBITED_MESSAGE = "Request is prohibited by organization's policy."
+
+
+@pytest.fixture
+def client():
+    return trace_v1.TraceServiceClient()
+
+
+@vpcsc_config.skip_unless_inside_vpcsc
+def test_list_traces_w_inside(client):
+    list(client.list_traces(vpcsc_config.project_inside))  # no perms issue
+
+
+@vpcsc_config.skip_unless_inside_vpcsc
+def test_list_traces_w_outside(client):
+    with pytest.raises(exceptions.PermissionDenied) as exc:
+        list(client.list_traces(vpcsc_config.project_outside))
+
+    assert exc.value.message == _VPCSC_PROHIBITED_MESSAGE
```

Comment:
Could these be changed to

```suggestion
    assert _VPCSC_PROHIBITED_MESSAGE in exc.value.message
```

Reply:
e47f6657a09